react/week2/src/App.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from "react";
+import TodoList from "./TodoList";
+
+import "./App.css";
+
+const todoList = [
+  {
+    id: 1,
+    description: "Get out of bed",
+    deadline: "2021-11-16T02:06",
+  },
+  {
+    id: 2,
+    description: "Brush teeth",
+    deadline: "2021-11-16T02:06",
+  },
+  {
+    id: 3,
+    description: "Eat breakfast",
+    deadline: "2021-11-16T02:06",
+  },
+];
+
+function App() {
+  const [description, setDescription] = useState([]);
+  const [deadline, setDeadline] = useState("");
+  const [list, setList] = useState(todoList);
+
+  function handleAddInput(desc, id) {
+    const newList = [...list];
+    newList
+      .map((item) => {
+        return item.id;
+      })
+      .indexOf(id);
+
+    setDescription(desc);
+  }
+
+  function deleteListItem(id) {
+    const newList = [...list];
+    const index = newList
+      .map((x) => {
+        return x.id;
+      })
+      .indexOf(id);
+    newList.splice(index, 1);
+    setList(newList);
+  }
+
+  function addNewTodo(newDescription, newDeadline) {
+    const newTodo = {
+      description: newDescription,
+      deadline: newDeadline,
+    };
+    setList((list) => [...list, newTodo]);
+    setDescription("");
+    setDeadline("");
+  }
+  const [timer, setTimer] = useState(0);
```

Comment:
States should always go at the top of the component. I can recommend extracting the timer component and making it's own to make it clearer for yourself